sdk/resourcemanager/Proto.Client/compute/VirtualMachineScaleSetRollingUpgradeOperations.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.ResourceManager.Compute;
+using Azure.ResourceManager.Core;
+
+namespace Proto.Compute
+{
+    /// <summary> A class representing the operations that can be performed over a specific VirtualMachineScaleSetRollingUpgrade. </summary>
+    public partial class VirtualMachineScaleSetRollingUpgradeOperations : SingletonOperationsBase<ResourceGroupResourceIdentifier, VirtualMachineScaleSetRollingUpgrade>
+    {
+        /// <summary> Initializes a new instance of VirtualMachineScaleSetRollingUpgradeOperations for mocking. </summary>
+        protected VirtualMachineScaleSetRollingUpgradeOperations()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VirtualMachineOperations"/> class.
+        /// </summary>
+        /// <param name="operation"> The client parameters to use in these operations. </param>
+        /// <param name="id"> The identifier of the resource that is the target of operations. </param>
+        protected internal VirtualMachineScaleSetRollingUpgradeOperations(OperationsBase operation, ResourceGroupResourceIdentifier id)
```

Comment:
This id starts to become semantically off.

OperationBase.Id = my identifier

where as this id is really my parent since i need to know my resourcegroupname.